README.md

```diff
@@ -57,48 +60,253 @@ This is work in progress as of 14 Nov 2018:
+
+```TypeScript
+import { endpoint, genericError, header, pathParam, queryParam, request, response, specificError } from "@airtasker/spot";
+
+class MyUserEndpoints {
+  @endpoint({
+    method: "GET",
+    path: "/users",
+    description: "Retrieve all users",
+    tags: ["Users"]
+  })
+  getUsers(@queryParam({ description: "Search term" }) search_term: string): UserResponse[] {
+    return response();
+  }
+
+  @endpoint({
+    method: "GET",
+    path: "/users/:id",
+    description: "Get user by id",
+    tags: ["Users"]
+  })
+  getUser(@pathParam({ description: "Unique user identifier" }) id: string): UserResponse {
+    return response();
+  }
+
+  @endpoint({
+    method: "POST",
+    path: "/users",
+    description: "Create a user",
+    tags: ["Users"]
+  })
+  @specificError<ApiError>({
+    name: "unauthorized",
+    statusCode: 401
+  })
+  createUser(
+    @request req: CreateUserRequest,
+    @header({
+      name: "Authorization",
+      description: "This is the authorization token"
+    })
+    authToken: Optional<string>
+  ): CreateUserResponse {
+    return response();
+  }
+}
+
+interface UserResponse {
+  firstName: string;
+  lastName: string;
+}
+
+interface CreateUserRequest {
+  firstName: string;
+  lastName: string;
+}
+
+interface CreateUserResponse {
+  success: boolean;
+}
+
+interface ApiError {
+  message: string;
+}
+```
+
+| Field                | Default            | Description                                               |
+| -------------------- | ------------------ | --------------------------------------------------------- |
+| `method`             |                    | (**required**) [HTTP method](#suppported-http-methods)    |
+| `path`               |                    | (**required**) URL path                                   |
+| `description`        | `""`               | Description of the endpoint                               |
+| `requestContentType` | `application/json` | Content type of the request body                          |
+| `successStatusCode`  | `200`              | HTTP status code for a successful response                |
+| `tags`               |                    | Array of tags used for endpoint grouping in documentation |
+
+### `@request`
+
+Define a request body for requests that require one. This is commonly used for `POST` and `PUT` requests:
+
+```TypeScript
+class MyUserEndpoints {
+  //...
+  @endpoint({
+    method: "POST",
+    path: "/users"
+  })
+  createUser(
+    @request req: CreateUserRequest
+  ): UserResponse {
+    return response();
+  }
+
+  @endpoint({
+    method: "PUT",
+    path: "/users/:id"
+  })
+  updateUser(
+    @pathParam({ description: "User unique identifier" }) id: string,
+    @request req: UpdateUserRequest
+  ): UserResponse {
+    return response();
+  }
+  //...
+}
+
+interface CreateUserRequest {
+  firstName: string;
+  lastName: string;
+}
+
+interface UpdateUserRequest {
+  lastName: string;
+}
+```
+
+### `@header`
```

Comment:
Would it be possible to add examples for each of these too, or mention that the examples above cover them?